File: bubble-sort/python/bubble_sort.py
def bubble_sort(the_list):
    if len(the_list) < 2:
        return the_list
    for i in range(len(the_list)):
        swapOccured = False
        for j in range(1, len(the_list) - i):
            if the_list[j] < the_list[j - 1]:
                temp = the_list[j]
                the_list[j] = the_list[j - 1]
                the_list[j - 1] = temp
                swapOccured = True
        if not swapOccured:
            break
    return the_list

a_list = [4, 6, 2, 3, 8, 5]
print(a_list)
bubble_sort(a_list)
print(a_list)

File: bubble-sort/javascript/bubbleSort.js
Array.prototype.bubbleSort = function(){
  if(this.length < 2){
    return this;
  }
  for(var i = 0; i<this.length; i++){
    var swapOccured = false;
    for(var j = 1; j<this.length - i; j++){
      if(this[j] < this[j-1]){
        var temp = this[j];
        this[j] = this[j-1];
        this[j-1] = temp;
        swapOccured = true;
      }
    }
    if(!swapOccured){
      break;
    }
  }
  return this;
}

var aList = [8, 3, 5, 1, 3, 9, 7];
aList.bubbleSort();
console.log(aList);
